File: ai/src/main/java/org/pingel/bayes/ModelGraph.java
package org.pingel.bayes;

import java.util.HashMap;
import java.util.Map;

import org.pingel.util.DirectedGraph;

public class ModelGraph extends DirectedGraph<RandomVariable, ModelEdge>
{
	Map<String, RandomVariable> name2variable = new HashMap<String, RandomVariable>();
	
	public RandomVariable addVertex(RandomVariable var)
	{
		return super.addVertex(var);
	}
	
	public RandomVariable getVariable(String name)
	{
		return name2variable.get(name);
	}
}

File: ai/src/main/scala/org/pingel/bayes/ModelGraph.scala
package org.pingel.bayes

import org.pingel.util.DirectedGraph
import org.pingel.util.DirectedGraphEdge

class ModelGraph extends DirectedGraph[RandomVariable, ModelEdge]
{
  def name2variable = Map[String, RandomVariable]()
	
  def addVertex(variable: RandomVariable): RandomVariable = {
    super.addVertex(variable)
  }
	
  def getVariable(name: String): RandomVariable = name2variable(name)
  
}

class ModelEdge(v1: RandomVariable, v2: RandomVariable)
extends DirectedGraphEdge[RandomVariable]
{
  def getSource = v1
  def getDest = v2
}

